src/main/java/solitaire/game/CardStack.java

```diff
@@ -55,10 +57,10 @@ public void addNewCard(int id) {
         solveFlipType(flipType);
     }
 
-    private void solveFlipType(int flipType) {
+    public void solveFlipType(int flipType) {
         switch (flipType) {
             case FLIPTYPE_TOP:
-                for (int i = 0; i < cards.size() - 1; i++) {
+                for (int i = 0; i < cards.size(); i++) {
```

Comment:
This is immediately undone down on line 66, therefore this change is bad.
```suggestion
                for (int i = 0; i < cards.size() - 1; i++) {
```